### ISSUE TITLE
'authenticated' event fired but authResult is undefined

### ISSUE THREAD

**Issue by TheNoim** [auth0/lock#700]
Otherwise, create a new issue with the following information:
-------------------------------------------------------------

- [ ] Code snippet or sample project that reproduces the bug:

```javascript
const auth0options = {
    allowSignUp: false,
    allowedConnections: ['github', 'google-oauth2', 'twitter','Username-Password-Authentication'],
    autoclose: true,
    auth: {
        sso: true
    }
};
var lock = new Auth0Lock(
    '',
    ''
,auth0options);
lock.on("authenticated", function(authResult) {
        console.log("authenticated!");
        var profile = JSON.parse(localStorage.getItem('profile'));
        // Only for testing.
        console.log(`PROFILE==> ${authResult}`);
        lock.getProfile(authResult.idToken, function(error, profile) {
            if (error) {
                console.error(error);
                return;
            }
            console.log(`PROFILE==> ${JSON.stringify(profile)}`);
            localStorage.setItem('idToken', authResult.idToken);
            localStorage.setItem('profile', JSON.stringify(profile));
        });
    });
```
- [ ] Screenshots when appropriate: http://imgur.com/a/gmuIK
- [ ] Lock version: 10.5.1
- [ ] Browser & OS: Phonegap -> iPhone SE Simulator newest Build
- [ ] Phonegap version: 6.3.4
- [ ] inAppBrowser Version: 1.5.0

**Comment by hzalaz** [comment]
@TheNoim you should use `sso: false` since it won't work in a phonegap app. Besides that, and assuming the console log in the screenshot is for authResult not for profile, could you check if any of the request returned an error in the console.

**Comment by wargun02** [comment]
HI All,

I 2nd this issue, running on lock.min.js version 10.5.1, with just sso: false as lock option. The 'authenticated' was fired after Facebook or Google+ registration and/or login, but the authResult is 'undefined'.

Yes, I have installed the InAppBrowser Plugin.

**Comment by hzalaz** [comment]
@wargun02 does any request fail or are blocked by a cordova plugin like the whitelist plugin, do you see any CORS issue?. If you can reproduce the issue with our sample project from https://docs.auth0.com please let us know.

**Comment by wargun02** [comment]
@hzalaz, I believe the CORS was set correctly with in Auth0.com's console as `file://\*` (for mobile). And on config.xml, I have both `<access origin="*"/> and <allow-navigation href="*"/>`.

Hang in there, just give me a moment, I'll be right back with some test results as per your suggestion on the sample project.

**Comment by krisklosterman** [comment]
I am using angular 2 using the guide here https://auth0.com/docs/quickstart/spa/angular2/01-login

I am also getting a return of undefined when authenticating.

```javascript
{
accessToken: ""
idToken: undefined
idTokenPayload: undefined
refreshToken: undefined 
}
```
Here is the auth.service.ts file I created:

```javascript
import { Injectable }      from '@angular/core';
import { tokenNotExpired } from 'angular2-jwt';

// Avoid name not found warnings
declare var Auth0Lock: any;

@Injectable()
export class Auth {
  // Configure Auth0
  lock = new Auth0Lock('Hidden', 'Hidden', {});

  constructor() {
    // Add callback for lock `authenticated` event
    this.lock.on("authenticated", (authResult) => {
      localStorage.setItem('id_token', authResult.idToken);
      console.log(authResult);
    });
  }

  public login() {
    // Call the show method to display the widget.
    this.lock.show();
  };

  public authenticated() {
    // Check if there's an unexpired JWT
    // This searches for an item in localStorage with key == 'id_token'
    return tokenNotExpired();
  };

  public logout() {
    // Remove token from localStorage
    localStorage.removeItem('id_token');
  };
}
```

**Comment by krisklosterman** [comment]
Any Ideas on this? I've tested the above code on 10.7.1

**Comment by josephbridgwaterrowe** [comment]
We're trying to use Auth0 with LinkedIn on Cordova and are having the same problems (authResult being undefined). I can provide further details if they would be useful.

**Comment by hzalaz** [comment]
This is probably caused by angular router trying to handle the OAuth2 response and dropping everything after the `#access_token` so Lock can't get the result (We are working to show an error in this case). A workaround is https://github.com/auth0-samples/auth0-angularjs2-systemjs-sample/issues/40#issuecomment-265170465 if you can't use a different router strategy besides hash for angular

**Comment by hzalaz** [comment]
Closing this one since it's related to angular router stripping the hash after the first string so `#access_token=TOKEN` will end up being `#access_token`.
